pkg/payload/task_graph.go

```diff
@@ -0,0 +1,521 @@
+package payload
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+	"sort"
+	"strconv"
+	"strings"
+	"sync"
+
+	"github.com/golang/glog"
+
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+)
+
+//
+// g := newGraph(tasks)
+// g.Split(OnJobs) -> jobs serialize the graph
+// g.Parallelize(ByNumberAndComponent) -> if a node has 0000_99_A_* and 0000_99_B_*, parallelize
+
+// task has type, name, grouping 0000_XX_yy-yy-yy_ORDERED
+// all groups with common XX and different yy-yy-yy may be executed in parallel (0000_99_cluster-monitoring-operator and 0000_99_cluster-samples-operator)
+// within a group:
+//   - all executable or unrecognized types form synchronization barriers (ClusterOperator, Deployment, Pod are all sync barriers)
+//   - all other types may be greedily created (Namespace, RoleBinding, Service, Route)
+// all jobs are global synchronization barriers
+//
+
+// SplitOnJobs enforces the rule that any Job in the payload prevents reordering or parallelism (either before or after)
+func SplitOnJobs(task *Task) bool {
+	return task.Manifest.GVK == schema.GroupVersionKind{Kind: "Job", Version: "v1", Group: "batch"}
```

Comment:
`k8s.io/api/batch/v1.SchemeGroupVersion.WithKind("Job")`

Reply:
I'd prefer not to take the dependency on the actual types when this is a constant that can never change anyway.